jsonrpc/helper.go

```diff
@@ -0,0 +1,191 @@
+package jsonrpc
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/0xPolygon/polygon-edge/state"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+type latestHeaderGetter interface {
+	Header() *types.Header
+}
+
+// GetNumericBlockNumber returns block number based on current state or specified number
+func GetNumericBlockNumber(number BlockNumber, store latestHeaderGetter) (uint64, error) {
+	switch number {
+	case LatestBlockNumber:
+		return store.Header().Number, nil
+
+	case EarliestBlockNumber:
+		return 0, nil
+
+	case PendingBlockNumber:
+		return 0, fmt.Errorf("fetching the pending header is not supported")
+
+	default:
+		if number < 0 {
+			return 0, fmt.Errorf("invalid argument 0: block number larger than int64")
```

Comment:
could we add number to the error message?

Reply:
You can see the error has at the latest commit
https://github.com/0xPolygon/polygon-edge/blob/1c181d3f8ebdb9db2d0742371ef85c814307eb52/jsonrpc/helper.go#L70